File: tests/nodejs/cases/case_206_thread_pool_capture_01.js
/**
 * Task case 206: thread_pool_capture_01 deep stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case206ThreadPoolCapture01(input) {
  const taskName = 'thread_pool_capture_01';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'extreme',
    stage: 'stress',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: async task captures payload.
  setTimeout(() => retainedAudit.push(payload), 0);
  return 'ok';
}

module.exports = { case206ThreadPoolCapture01 };

File: tests/go/cases/case_206_thread_pool_capture_01.go
package escape_tests

var retainedCase206 = []map[string]string{}

func Case206ThreadPoolCapture01(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "thread_pool_capture_01",
		"entity": "extreme",
		"stage": "stress",
		"input": raw,
		"checksum": "thread_pool_capture_01:" + raw,
	}
	// ESCAPE: goroutine captures payload and retains it.
	go func(v map[string]string) { retainedCase206 = append(retainedCase206, v) }(payload)
	return "ok"
}
